### PR TITLE
Implement `formatter` specializations for container adaptors

### LINKED ISSUE #2919: P2286R8 Formatting Ranges
WG21-P2286R8 Formatting Ranges
WG21-P2585R1 Improve Default Container Formatting
WG21-P2713R1 Escaping Improvements In `std::format`
LWG-3631 `basic_format_arg(T&&)` should use `remove_cvref_t<T>` throughout (affects `formattable` concept)
LWG-3750 Too many papers bump `__cpp_lib_format`
LWG-3839 `range_formatter`'s `set_separator`, `set_brackets`, and `underlying` functions should be `noexcept`
LWG-3881 Incorrect formatting of container adaptors backed by `std::string`
LWG-3892 Incorrect formatting of nested ranges and tuples
LWG-3925 Concept `formattable`'s definition is incorrect

Feature-test macro:
```cpp
#define __cpp_lib_format_ranges 202207L
```

### PR DESCRIPTION
**Pull request by frederick-vs-ja**
Completes WG21-P2286R8. Implements LWG-3881 and LWG-3750. Fixes #2919.

Unblocks libcxx tests:
- `std/containers/container.adaptors/container.adaptors.format/format.functions.format.pass.cpp`
- `std/containers/container.adaptors/container.adaptors.format/format.functions.vformat.pass.cpp`
- `std/containers/container.adaptors/container.adaptors.format/parse.pass.cpp`
- `std/containers/container.adaptors/container.adaptors.format/types.compile.pass.cpp`
- `std/utilities/format/format.formattable/concept.formattable.compile.pass.cpp`

### CODE REVIEW

File: stl/inc/queue [hunk 1]
```diff
@@ -18,6 +18,7 @@
 
 #if _HAS_CXX23
 #include <__msvc_ranges_to.hpp>
+#include <format>
```
Comment (frederick-vs-ja) [inline stl/inc/queue:21]: Currently `<queue>` and `<stack>` are made to include `<format>`.
I'll make a follow-up PR to reduce inclusion from `<queue>`, `<stack>`, `<stacktrace>`, and `<thread>`.

File: tests/std/tests/P2286R8_text_formatting_container_adaptors/test.cpp [hunk 2]
```diff
@@ -0,0 +1,1206 @@
...
+
+    check_exception(
+        "The argument index value is too large for the number of arguments supplied", SV("{:^^{}::>5}"), input);
+    check_exception(
+        "The argument index value is too large for the number of arguments supplied", SV("{:^^{}::>{}}"), input, 12);
+}
+
+template <class CharT, class TestFunction, class ExceptionTest>
+void test_pointer(TestFunction check, ExceptionTest check_exception) {
+    array input{static_cast<void*>(0)};
```
Comment (StephanTLavavej) [inline tests/std/tests/P2286R8_text_formatting_container_adaptors/test.cpp:954]: This should be `nullptr`, we forbid zero-as-null in our codebase.

File: tests/std/tests/P2286R8_text_formatting_container_adaptors/test.cpp [hunk 3]
```diff
@@ -0,0 +1,1206 @@
...
+// * std/containers/container.adaptors/container.adaptors.format/format.functions.format.pass.cpp
+// * std/containers/container.adaptors/container.adaptors.format/format.functions.vformat.pass.cpp
+
+#include <algorithm>
+#include <array>
+#include <cassert>
+#include <charconv>
+#include <concepts>
+#include <format>
+#include <list>
```
Comment (StephanTLavavej) [inline tests/std/tests/P2286R8_text_formatting_container_adaptors/test.cpp:24]: `<list>` is unused. Conversely, we should directly include the following:

```cpp
#include <cctype> // toupper
#include <cstddef> // size_t
#include <cstdint> // uint16_t
#include <cstring> // strlen
#include <functional> // greater
#include <iterator> // back_inserter
#include <locale> // locale
#include <string_view> // basic_string_view
#include <string> // basic_string
#include <type_traits> // type_identity_t
#include <utility> // forward
#include <vector> // vector
```

Some are included by `<test_format_support.hpp>`, but having tests directly include what they use makes it easier to refactor support headers.

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for completing this major, long-awaited C++23 feature!

# :heart_eyes_cat: :tada: :partying_face: :green_heart:

**Pull request merged**